src/SFML/Window/WindowBase.cpp

```diff
@@ -207,18 +211,67 @@ void WindowBase::setSize(const Vector2u& size)
 {
     if (m_impl)
     {
-        m_impl->setSize(size);
+        // Constrain requested size within minimum and maximum bounds
+        const auto minimumSize = m_impl->getMinimumSize().value_or(sf::Vector2u());
+        const auto maximumSize = m_impl->getMaximumSize().value_or(
+            sf::Vector2u(std::numeric_limits<unsigned int>::max(), std::numeric_limits<unsigned int>::max()));
+        const auto width  = std::clamp(size.x, minimumSize.x, maximumSize.x);
+        const auto height = std::clamp(size.y, minimumSize.y, maximumSize.y);
+
+        // Do nothing if requested size matches current size
+        const sf::Vector2u clampedSize(width, height);
+        if (clampedSize == m_size)
+            return;
+
+        m_impl->setSize(clampedSize);
 
         // Cache the new size
-        m_size.x = size.x;
-        m_size.y = size.y;
+        m_size = clampedSize;
 
         // Notify the derived class
         onResize();
     }
 }
 
 
+////////////////////////////////////////////////////////////
+void WindowBase::setMinimumSize(const std::optional<Vector2u>& minimumSize)
+{
+    if (m_impl)
+    {
+        [[maybe_unused]] const auto validateMinimumSize = [this, minimumSize]()
+        {
+            if (!minimumSize.has_value() || !m_impl->getMaximumSize().has_value())
+                return true;
+            return minimumSize->x <= m_impl->getMaximumSize()->x && minimumSize->y <= m_impl->getMaximumSize()->y;
+        };
+        assert(validateMinimumSize() && "Minimum size cannot be bigger than the miximum size along either axis");
+
+        m_impl->setMinimumSize(minimumSize);
+        setSize(getSize());
+    }
+}
```

Comment:
If the current window style was stored within the class, we could return early if this function or `setMaximumSize` is called when the style include `sf::Style::Fullscreen` or excludes `sf::Style::Resize`. That would let me remove some fixes I pushed to the X11 implementation and help ensure all implementations always behave the same.